solr/core/src/java/org/apache/solr/update/processor/DistributedZkUpdateProcessor.java

```diff
@@ -314,6 +314,23 @@ public void processDelete(DeleteUpdateCommand cmd) throws IOException {
 
   @Override
   protected void doDeleteById(DeleteUpdateCommand cmd) throws IOException {
+
+    // if using the CompositeId router and route field is missing, distribute to all shard leaders
+    if(cmd.getRoute()==null) {
+      zkCheck();
+      DocCollection coll = zkController.getClusterState().getCollection(collection);
+      DocRouter router = coll.getRouter();
+      String routeField = router.getRouteField(coll);
+      if (router instanceof CompositeIdRouter && routeField != null)  {
+        DistribPhase phase = DistribPhase.parseParam(req.getParams().get(DISTRIB_UPDATE_PARAM));
+        if (phase == DistribPhase.NONE) {
+          log.debug("Using compositeId router and deleteById command is with missing route value, distributing to all shard leaders");
```

Comment:
@makosten small nit, should this debug line be "command is missing" instead of "is with" ;-)